scripts/install-nix-from-closure.sh

```diff
@@ -1,149 +1,618 @@
-    exit 1
-fi
-
-# Install an SSL certificate bundle.
-if [ -z "$NIX_SSL_CERT_FILE" ] || ! [ -f "$NIX_SSL_CERT_FILE" ]; then
-    $nix/bin/nix-env -i "$cacert"
-    export NIX_SSL_CERT_FILE="$HOME/.nix-profile/etc/ssl/certs/ca-bundle.crt"
-fi
-
-# Subscribe the user to the Nixpkgs channel and fetch it.
-if ! $nix/bin/nix-channel --list | grep -q "^nixpkgs "; then
-    $nix/bin/nix-channel --add https://nixos.org/channels/nixpkgs-unstable
-fi
-if [ -z "$_NIX_INSTALLER_TEST" ]; then
-    $nix/bin/nix-channel --update nixpkgs
-fi
-
-added=
-if [ -z "$NIX_INSTALLER_NO_MODIFY_PROFILE" ]; then
-
-    # Make the shell source nix.sh during login.
-    p=$HOME/.nix-profile/etc/profile.d/nix.sh
-
-    for i in .bash_profile .bash_login .profile; do
-        fn="$HOME/$i"
-        if [ -w "$fn" ]; then
-            if ! grep -q "$p" "$fn"; then
-                echo "modifying $fn..." >&2
-                echo "if [ -e $p ]; then . $p; fi # added by Nix installer" >> "$fn"
+
+    if [ -d "$dest" ]; then
+        print "Directory $dest already exists. Skipping creation of $dest"
+    fi
+
+}
+
+installNix() {
+
+    # In this function, if some action makes any change to the system, or does err Exit - it's need to be wrapped in a function, ex func(){...}. Inside func() you start with setup of traps, before errors/actions happen, so and according revert function be launched: 'trap funcRevertOnCondition condition', and funcRevertOnCondition is described right after trap.
+    # How all actions operate inside this function:
+    #**************************************************
+    #
+    #    installNix() {
+    #
+    #        func() {
+    #
+    #            trap 'funcRevertOnCondition $LINENO $?' condition
+    #            funcRevertOnCondition() {
+    #                errorRevert "Received Error signal from line $1"
+    #                # revert action
+    #                >&2 contactme # This function gives contact information
+    #                exit "$?"
+    #            }
+    #
+    #            # action of functions
+    #            # 'exit SING # to the trap that catches exit SIGN'
+    #        }
+    #
+    #        func
+    #
+    #    }
+    #
+    #**************************************************
+
+    # Function must have revert procedure. In case of any problem - function quits without artifacts/changes to the system
+    # Nix is pure, and claims to be transnational, so let's show people clean transnational installation
+
+    createDest() {
+
+        trap destCreateRevert 1
+
+        destCreateRevert() {
+            # all means failed, Exit!
+            error "
+
+    Can't get root access to create $dest folder. All variants failed.
+    Exiting...
+    "
+        }
+
+        # Shell parses variable with current user, so value under sudo/su not changes.
+        # It is const, but don't move it above USER set/checks.
+        readonly cmd="mkdir -m 0755 $dest && chown $USER $dest"
```

Comment:
Is there a possibility that `$dest` includes spaces? If so, it would need to be properly quoted.

Reply:
$dest is quoted inside "" already. This means spaces is processed correctly.

Time before, I had style to do:
```shell
readonly cmd='mkdir -m 0755 '"$dest"' && chown '"$USER"' '"$dest"
```

But now I consider that very redundant. It is not readable or understandable, and additional hassle to maintain. And may be less performance efficient.

It is processes spaces properly. Or I don't understand something.

Reply:
No, you're passing this command to sudo and sh with the `-c` flag, and they pass that to a subshell where it gets interpreted again. If you want to do it properly you need to quote them, like this: https://github.com/wmertens/sudo.bash/blob/master/sudo.bash#L81-L87

Reply:
Yes. Right. Even when I wrote that I thought probably those quotes not save me from using command further.

I used escaping like that before, but I will go out of my way to not to use it, since it is unreadable magic loop.

To look how to solve it differently I will get timeout on this one.
I know this is classic issue, but. Maybe )

Reply:
This problem still exist and not accounted for in the master also.

This is upstream issue I just not fixing in this PR.